src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -679,14 +726,10 @@ mongoc_cluster_run_command_private (mongoc_cluster_t *cluster,
    if (!reply) {
       reply = &reply_local;
    }
-   server_stream = cmd->server_stream;
-   if (server_stream->sd->max_wire_version >= WIRE_VERSION_OP_MSG) {
-      retval = mongoc_cluster_run_opmsg (cluster, cmd, reply, error);
-   } else {
-      retval =
-         mongoc_cluster_run_command_opquery (cluster, cmd, -1, reply, error);
-   }
-   _handle_not_primary_error (cluster, server_stream, reply);
+
+   retval = _mongoc_cluster_run_command_selected_mech_private(cluster, cmd, reply, error);
+
+    _handle_not_primary_error (cluster, server_stream, reply);
```

Comment:
One possible way to validate these changes is to use a mock server test. Mock server tests are a blunt instrument, but I do not see another easy alternative to test.

Here is an example of a mock server test validating that `isMaster` is the first command sent:
https://github.com/mongodb/mongo-c-driver/commit/aa76669d60e3bdc9670eedcf64bc12472bd03ec8#diff-d35f0d63a4ad8bfb630b204735bbc00dab445deed4a03e0e23793671c5fb4152R26